redaxo/src/core/lib/sql/sql.php

```diff
@@ -1463,4 +1501,53 @@ public static function checkDbConnection($host, $login, $pw, $dbname, $createDb
 
         return  $err_msg;
     }
+
+    private function setMultiRecordQuery($verb)
+    {
+        $fields = [];
+
+        foreach ($this->records as $record) {
+            foreach ($record->values as $field => $value) {
+                $fields[$field] = true;
+            }
+            foreach ($record->rawValues as $field => $value) {
+                $fields[$field] = true;
+            }
+        }
+
+        $fields = array_keys($fields);
+
+        $rows = [];
+        $params = [];
+
+        foreach ($this->records as $record) {
```

Comment:
Wenn man hier große text/blob spalten hat, kann man hier leicht out-of-memory laufen.

Ich hab sowas schon öfter mit 500 records per batch ganz gut gehandelt.
Es gibt auch einen mysql buffer der diese max. Bytes für multi-row inserts limitiert.

Ausserdem hab ich die Erfahrung gemacht, dass der Performance-gewinn nicht so stark von der Batch Größe abhängt.

Reply:
Bin mir nicht sicher, ob ich deinen Kommentar richtig interpretiere.
Ist das nur ein Hinweis, dass man aufpassen muss, wenn man sehr viele Zeilen einfügen möchte, und man die dann ggf. in Batches teilen sollte.
Oder würdest du hier in der Methode was ändern?

Reply:
Ich habe die Erfahrung gemacht dass man hier eine Fehlerquelle hat und ggf. Zu große inserts dür die db baut. Dies habe ich in der vergangenheit durch 500 row batching gehandelt.

Ich denke in dieses problem werden wir hier früher oder später auch laufen und sollten wir ggf.. jetzt schon behandeln

Reply:
Ich hab sowas immer mit einer array-chunk schleife aussen herum behandelt.

Reply:
Bloß wie? out-of-memory, da meinst du ja vermutlich php-seitig, oder?
Und um das zu verhinden, muss der Caller ja eher schon die `addRecord`-Aufrufe in Batches teilen, und am Ende jedes Batches `insert()` aufrufen.
Wenn wir hier in der Methode in 500er Pakete teilen, sind ja trotzdem alle Records im Speicher und wir haben (php-seitig) nichts gewonnen, oder?

Allerdings Limits db-seitig würden wir damit schon ggf. umgehen.

Reply:
Ist es sinnvoll, die Chunks dann alle in einer Transaction abzusetzen, oder gibt es da auch Limits, die wir dann überschreiten könnten?

Reply:
Wir nutzen in rex core bisher nirgends Transaktionen, oder?

Reply:
Bisher nicht, mit der 5.6 wird die Api dazu aber ja eingeführt:
https://github.com/redaxo/redaxo/pull/1599

(In YForm werden wir die wahrscheinlich dann verwenden.)

Reply:
Hmm, bin am überlegen, ob man das mit den Chunks nicht vielleicht doch erst noch mal ganz weglassen soll?
Bei Bedarf kann man ja auch relativ leicht "von außen" die Datensätze in Chunks teilen. Und hat dann aber auch selbst die Wahl, ob man dabei Transactions nutzt oder nicht.

Reply:
Auch ok